File: bindings/csharp/src/ProvenCoap.cs
// SPDX-License-Identifier: MPL-2.0
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// CoAP protocol types for proven-servers.

namespace Proven;

/// <summary>Method matching the Idris2 ABI tags (0-3).</summary>
public enum Method : byte
{
    Get = 0,
    Post = 1,
    Put = 2,
    Delete = 3
}

/// <summary>MessageType matching the Idris2 ABI tags (0-3).</summary>
public enum MessageType : byte
{
    Confirmable = 0,
    NonConfirmable = 1,
    Acknowledgement = 2,
    Reset = 3
}

/// <summary>ContentFormat matching the Idris2 ABI tags (0-6).</summary>
public enum ContentFormat : byte
{
    TextPlain = 0,
    LinkFormat = 1,
    Xml = 2,
    OctetStream = 3,
    Exi = 4,
    Json = 5,
    Cbor = 6
}

/// <summary>ResponseClass matching the Idris2 ABI tags (0-4).</summary>
public enum ResponseClass : byte
{
    Success = 0,
    ClientError = 1,
    ServerError = 2,
    Signaling = 3,
    Empty = 4
}

/// <summary>SessionState matching the Idris2 ABI tags (0-4).</summary>
public enum SessionState : byte
{
    Idle = 0,
    Bound = 1,
    Serving = 2,
    Observing = 3,
    Shutdown = 4
}

File: bindings/php/src/ProvenCoap.php
<?php
// SPDX-License-Identifier: MPL-2.0
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// CoAP protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** Method matching the Idris2 ABI tags. */
enum Method: int
{
    case Get = 0;
    case Post = 1;
    case Put = 2;
    case Delete = 3;
}

/** MessageType matching the Idris2 ABI tags. */
enum MessageType: int
{
    case Confirmable = 0;
    case NonConfirmable = 1;
    case Acknowledgement = 2;
    case Reset = 3;
}

/** ContentFormat matching the Idris2 ABI tags. */
enum ContentFormat: int
{
    case TextPlain = 0;
    case LinkFormat = 1;
    case Xml = 2;
    case OctetStream = 3;
    case Exi = 4;
    case Json = 5;
    case Cbor = 6;
}

/** ResponseClass matching the Idris2 ABI tags. */
enum ResponseClass: int
{
    case Success = 0;
    case ClientError = 1;
    case ServerError = 2;
    case Signaling = 3;
    case Empty = 4;
}

/** SessionState matching the Idris2 ABI tags. */
enum SessionState: int
{
    case Idle = 0;
    case Bound = 1;
    case Serving = 2;
    case Observing = 3;
    case Shutdown = 4;
}
